File: cgi/test_cgi.php
#!/usr/local/bin/php
<?php

// Minimal CGI PHP script

// CGI headers
//header("Content-Type: text/html");

// Body
echo "<html>";
echo "<head><title>CGI Test (PHP)</title><link rel=\"icon\" href=\"data:,\"></head>";
echo "<body>";
echo "<h1>Hello from PHP CGI!</h1>";
// echo "<p>REQUEST_METHOD = " . $_SERVER['REQUEST_METHOD'] . "</p>";
// echo "<p>SCRIPT_NAME = " . $_SERVER['SCRIPT_NAME'] . "</p>";
echo "CWD: " . getcwd() . "\n"; 
echo "<p>QUERY_STRING = " . $_SERVER['QUERY_STRING'] . "</p>";
// echo "<p>SERVER_PROTOCOL = " . $_SERVER['SERVER_PROTOCOL'] . "</p>";
echo "</body></html>";
?>

File: cgi/test_cgi.py
#!/usr/bin/env python3

import os

# CGI requires printing HTTP headers first
#print("Content-Type: text/html\n")

print("<html>")
print("<head><title>CGI Test</title></head>")
print("<body>")
print("<h1>Hello from Python CGI!</h1>")

# Show some environment variables (CGI passes these)
#print("<h2>Environment Variables</h2>")
#print("<ul>")
#for key, value in os.environ.items():
   # print(f"<li>{key} = {value}</li>")
#print("</ul>")

print("</body></html>")
